src/dispatch/signal/service.py

```diff
@@ -238,100 +209,52 @@ def create_instance(*, db_session, signal_instance_in: SignalInstanceCreate) ->
-
-    # use tags if we have them
-    if duplication_rule:
-        if signal_instance.tags:
-            tag_type_names = [t.name for t in duplication_rule.tag_types]
-            hash_values = []
-            for tag in signal_instance.tags:
-                if tag.tag_type.name in tag_type_names:
-                    hash_values.append(tag.tag_type.name)
-            fingerprint = hashlib.sha1("-".join(sorted(hash_values)).encode("utf-8")).hexdigest()
-
-    return fingerprint
-
-
-def deduplicate(
-    *, db_session, signal_instance: SignalInstance, duplication_rule: DuplicationRule
-) -> bool:
-    """Find any matching duplication rules and match signals."""
-    duplicate = False
-
-    # always fingerprint
-    fingerprint = create_instance_fingerprint(duplication_rule, signal_instance)
-    signal_instance.fingerprint = fingerprint
-    db_session.commit()
-
-    if not duplication_rule:
-        return duplicate
-
-    if duplication_rule.mode != RuleMode.active:
-        return duplicate
-
-    window = datetime.now(timezone.utc) - timedelta(seconds=duplication_rule.window)
-    fingerprint = create_instance_fingerprint(duplication_rule.tag_types, signal_instance)
+def apply_filter_actions(*, db_session, signal_instance: SignalInstance):
+    """Applies any matching filter actions associated with this instance."""
 
-    instances = (
-        db_session.query(SignalInstance)
-        .filter(Signal.id == signal_instance.signal.id)
-        .filter(SignalInstance.id != signal_instance.id)
-        .filter(SignalInstance.created_at >= window)
-        .filter(SignalInstance.fingerprint == fingerprint)
-        .all()
-    )
-
-    if instances:
-        duplicate = True
-        # TODO find the earliest created instance
-        signal_instance.case_id = instances[0].case_id
-        signal_instance.duplication_rule_id = duplication_rule.id
-
-    db_session.commit()
-    return duplicate
-
-
-def supress(
-    *, db_session, signal_instance: SignalInstance, suppression_rule: SuppressionRule
-) -> bool:
-    """Find any matching suppression rules and match instances."""
-    supressed = False
-
-    if not suppression_rule:
-        return supressed
+    for f in signal_instance.signal.filters:
+        if f.mode != SignalFilterMode.active:
+            continue
 
-    if suppression_rule.mode != RuleMode.active:
-        return supressed
-
-    if suppression_rule.expiration:
-        if suppression_rule.expiration <= datetime.now():
-            return supressed
-
-    rule_tag_ids = sorted([t.id for t in suppression_rule.tags])
-    signal_tag_ids = sorted([t.id for t in signal_instance.tags])
-
-    if rule_tag_ids == signal_tag_ids:
-        supressed = True
-        signal_instance.suppression_rule_id = suppression_rule.id
-
-    db_session.commit()
-    return supressed
+        query = db_session.query(SignalInstance).filter(
+            SignalInstance.signal_id == signal_instance.signal_id
+        )
```

Comment:
```suggestion
        query = (
            db_session.query(SignalInstance)
            .join(SignalInstance.entities)
            .join(EntityType)
            .filter(SignalInstance.signal_id == signal_instance.signal_id)
        )
```

Resolves:

```bash
ERROR:The query does not contain model `EntityType`.:/Users/wshel/Projects/dispatch/src/dispatch/signal/scheduled.py:consume_signals:48
Traceback (most recent call last):
  File "/Users/wshel/Projects/dispatch/src/dispatch/signal/scheduled.py", line 41, in consume_signals
    signal_flows.create_signal_instance(
  File "/Users/wshel/Projects/dispatch/src/dispatch/signal/flows.py", line 41, in create_signal_instance
    if signal_service.apply_filter_actions(db_session=db_session, signal_instance=signal_instance):
  File "/Users/wshel/Projects/dispatch/src/dispatch/signal/service.py", line 233, in apply_filter_actions
    query = apply_filters(query, f.expression, model_cls=SignalInstance)
  File "/Users/wshel/Projects/dispatch/src/dispatch/database/service.py", line 319, in apply_filters
    sqlalchemy_filters = [filter.format_for_sqlalchemy(query, default_model) for filter in filters]
  File "/Users/wshel/Projects/dispatch/src/dispatch/database/service.py", line 319, in <listcomp>
    sqlalchemy_filters = [filter.format_for_sqlalchemy(query, default_model) for filter in filters]
  File "/Users/wshel/Projects/dispatch/src/dispatch/database/service.py", line 149, in format_for_sqlalchemy
    *[filter.format_for_sqlalchemy(query, default_model) for filter in self.filters]
  File "/Users/wshel/Projects/dispatch/src/dispatch/database/service.py", line 149, in <listcomp>
    *[filter.format_for_sqlalchemy(query, default_model) for filter in self.filters]
  File "/Users/wshel/Projects/dispatch/src/dispatch/database/service.py", line 118, in format_for_sqlalchemy
    model = get_model_from_spec(filter_spec, query, default_model)
  File "/Users/wshel/.pyenv/versions/3.9.7/envs/dispatch/lib/python3.9/site-packages/sqlalchemy_filters/models.py", line 114, in get_model_from_spec
    raise BadSpec(
sqlalchemy_filters.exceptions.BadSpec: The query does not contain model `EntityType`
```

Reply:
This should be covered by `query = apply_filter_specific_joins(SignalInstance, f.expression, query)` I'd be interested to know when you encountered this error?

Reply:
The error was raised when I tested signal ingestion end-to-end w/ a filter applied, using the scheduled consumer task.